src/Backups/BackupEntriesCollector.cpp

```diff
@@ -567,17 +567,16 @@ std::vector<std::pair<ASTPtr, StoragePtr>> BackupEntriesCollector::findTablesInD
 
     checkIsQueryCancelled();
 
-    auto filter_by_table_name = [my_database_info = &database_info](const String & table_name)
+    auto filter_by_table_name = [&](const String & table_name)
     {
-        /// We skip inner tables of materialized views.
-        if (table_name.starts_with(".inner_id."))
```

Comment:
The problem was here. We shouldn't write the metadata of a materialized view's internal table to a backup, and we shouldn't read it during `RESTORE` because a materialized view can create its internal table by itself. This code only handled the `.inner_id.` prefix to detect internal tables but not the `.inner.` prefix which is used in an Ordinary database. As a result during `RESTORE` it could try to create the `.inner.mv` table first, and then `mv` (the materialized view itself), which failed in the constructor of the materialized view with an error message that the `.inner.mv` table already exists.